src/app/app-components/courses/add-course/add-course.component.html

```diff
@@ -1,22 +1,72 @@
 <div class="add-course">
     <app-breadcrumbs-navbar></app-breadcrumbs-navbar>
-    <div class="add-course-container">
-        <div class="container">
-            <span>Title:</span>
-            <input [(ngModel)]="name" type="text" placeholder="Add title">
-            <span>Description:</span>
-            <textarea [(ngModel)]="description" placeholder="Add description"></textarea>
-            <span>Date:</span>
-            <app-date-input-component (dateChange)="onDateChange($event)" [date]="date"></app-date-input-component>
-            <span>Duration:</span>
-            <app-duration-input-component (durationChange)="onDurationChange($event)" [minutes]="length">
-            </app-duration-input-component>
-            <span>Authors:</span>
-            <input type="text" placeholder="Oops! this is temporarily disabled..." disabled>
-            <div class="actions">
-                <button (click)="onSave()" class="save-button" [disabled]="!enableSaveButton()">Save</button>
-                <button (click)="onCancel()" class="cancel-button">Cancel</button>
+    <form [formGroup]="formGroup">
+        <div class="add-course-container">
+            <div class="container">
+                <div class="row input-label" [class.error]="computeControlValidity(nameControl)">
+                    <span>Title:</span>
+                    <ng-container *ngIf="computeControlValidity(nameControl)">
+                        <span *ngIf="nameControl.hasError('required')">*required</span>
```

Comment:
We could create `get` properties to hold the logic of all `*ngIf`s. This allows for better separation of concerns between template and the component code.

Reply:
Sounds good